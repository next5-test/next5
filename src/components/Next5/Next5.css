.next5 {
  margin-top: 10px;
}

.next5 .header {
  border-radius: 6px 6px 0 0;
  -moz-border-radius: 6px 6px 0 0;
  -webkit-border-radius: 6px 6px 0 0;
  background: #252525;
  line-height: 40px;
  text-transform: uppercase;
  font-weight: 400;
  color: #fff;
  padding-left: 10px;
}
  
.next5 .content {
  background: #f0efee;
  border-radius: 0 0 6px 6px;
  -moz-border-radius: 0 0 6px 6px;
  -webkit-border-radius: 0 0 6px 6px;
  border-bottom: 1px solid #e2e2e2;
  padding: 5px;
}
  
.next5 table {
  width: 100%;
  cursor: pointer;
}

.next5 td {
  background-color: #f2f1f1;
  color: #252525;
  border-bottom: 1px solid #e8e7e7;
  border-top: 1px solid #fff;
  line-height: 35px;
}

.next5 tr:nth-child(odd) td {
  background: #ECECEC;
}

.next5 tr:hover td {
  background-color: #fafafa;
  -webkit-transition: background-color 200ms linear;
  -moz-transition: background-color 200ms linear;
  -o-transition: background-color 200ms linear;
  -ms-transition: background-color 200ms linear;
  transition: background-color 200ms linear;
}

.time{
  padding-left: 10px;
}
  