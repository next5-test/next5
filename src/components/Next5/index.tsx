export { default } from './Next5';