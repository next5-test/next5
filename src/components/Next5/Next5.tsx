import * as React from 'react';
import './Next5.css';
import { RaceType } from '../../types';
import { formatTimeRemaining, getRaceTypeName } from '../../utils';

interface Next5Props {
  races: RaceType[];
  time: number;
  onClick: Function;
}

const Next5 = (props: Next5Props) => {
  const { races, time, onClick } = props;

  return (
    <div className="next5">
      <div className="header">
        Next 5 Races
        </div>
      <div className="content">
        <table cellSpacing="0" cellPadding="0">
          <tbody>
            {races.map((race: RaceType) => (
              <tr
                key={`${race.EventID}-${race.RaceNum}`}
                onClick={() => onClick(race.EventID, race.Description)}
              >
                <td className="time">{formatTimeRemaining(race.SuspendTime - time)}</td>
                <td className="race">Race {race.RaceNum}</td>
                <td className="description">{race.Description}</td>
                <td className="type">{getRaceTypeName(race.RaceType)}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default Next5;