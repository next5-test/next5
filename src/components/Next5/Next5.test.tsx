import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Next5 from './Next5';

test('<Next5> renders OK', () => {
  const component = renderer.create(<Next5 races={[]} time={new Date().getTime()} onClick={() => null} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});