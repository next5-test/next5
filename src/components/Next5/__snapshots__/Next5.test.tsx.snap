// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Next5> renders OK 1`] = `
<div
  className="next5"
>
  <div
    className="header"
  >
    Next 5 Races
  </div>
  <div
    className="content"
  >
    <table
      cellPadding="0"
      cellSpacing="0"
    >
      <tbody />
    </table>
  </div>
</div>
`;
